[package] skip failed __import__ statements

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59591 [package] skip failed __import__ statements**
* #59326 [package] PackageExporter verbose mode print package contents
Change our dependency resolution AST parsing code to skip `__import__` statements that are unresolvable. Additionally adds a user warning to the PackageExporter when the verbose mode is set to 'True'.

Example of package verbose mode being printed out (now prints out locations of invalid imports):
```
Package Contents:
- interned modules:
    foo__
- explicitly externed modules:
- implicitly externed modules:
    builtins
    collections
- mocked modules:
- pickles:
    <dcgan.model.pkl>
- invalid objects:
    models.DCGAN
    models.loss_criterions.ac_criterion
    models.loss_criterions.base_loss_criterions
    sentencepiece
- location of unresolvable and skipped __import__ statements:
    module: 'foo__', lineno: 8
```

Differential Revision: [D28945833](https://our.internmc.facebook.com/intern/diff/D28945833)

Commit:
[package] skip failed __import__ statements

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28945833).

Lilyjjo commented:
@SplitInfinity I accidentally messed up the ghstack, but I addressed the comments you left on: https://github.com/pytorch/pytorch/pull/59330

Commit:
Update on "[package] skip failed __import__ statements"

Change our dependency resolution AST parsing code to skip `__import__` statements that are unresolvable. Additionally adds a user warning to the PackageExporter when the verbose mode is set to 'True'.

Example of package verbose mode being printed out (now prints out locations of invalid imports):
```
Package Contents:
- interned modules:
    foo__
- explicitly externed modules:
- implicitly externed modules:
    builtins
    collections
- mocked modules:
- pickles:
    <dcgan.model.pkl>
- invalid objects:
    models.DCGAN
    models.loss_criterions.ac_criterion
    models.loss_criterions.base_loss_criterions
    sentencepiece
- location of unresolvable and skipped __import__ statements:
    module: 'foo__', lineno: 8
```

Differential Revision: [D28945833](https://our.internmc.facebook.com/intern/diff/D28945833)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28945833).

Commit:
Update on "[package] skip failed __import__ statements"

Change our dependency resolution AST parsing code to skip `__import__` statements that are unresolvable. Additionally adds a user warning to the PackageExporter when the verbose mode is set to 'True'.

Example of package verbose mode being printed out (now prints out locations of invalid imports):
```
Package Contents:
- interned modules:
    foo__
- explicitly externed modules:
- implicitly externed modules:
    builtins
    collections
- mocked modules:
- pickles:
    <dcgan.model.pkl>
- invalid objects:
    models.DCGAN
    models.loss_criterions.ac_criterion
    models.loss_criterions.base_loss_criterions
    sentencepiece
- location of unresolvable and skipped __import__ statements:
    module: 'foo__', lineno: 8
```

Differential Revision: [D28945833](https://our.internmc.facebook.com/intern/diff/D28945833)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28945833).
